src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -114,6 +114,23 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
                         Return
                     End If
 
+                    If Not isExplicitFormat Then
+                        ' Avoid automatically formatting extremely large dirty spans without an explicit request from
+                        ' the user. The "large span threshold" is 7000 lines. The 7000 line threshold is an estimated
+                        ' value accounting for a lower-bound of the algorithmic complexity of text differencing in
+                        ' designer cases along with measurements of a pathological example demonstrated at 14000 lines.
+                        ' We expect Windows Forms designer formatting operations to run in under ~15 seconds on average
+                        ' current hardware when nearing the threshold.
+                        Dim startLineNumber = 0
+                        Dim startColumnIndex = 0
+                        Dim endLineNumber = 0
+                        Dim endColumnIndex = 0
+                        info.SpanToFormat.GetLinesAndColumns(startLineNumber, startColumnIndex, endLineNumber, endColumnIndex)
+                        If endLineNumber - startLineNumber > 7000 Then
```

Comment:
Use an option, in case a customer needs a workaround and we can just point them at a reg key to set?

Reply:
It wasn't an option in the past, and this PR is targeting a bug submitted by a customer upgrade scenario. An option seems like unnecessary complexity given multiple alternatives which have more desirable overall outcomes. This limit covers all reported cases of this bug with certainty, without affecting other users.

Reply:
regkey option is not exposed to users. so this kind of arbitrary restriction we put, we usually have a regkey option so that one can tweak if they want to.

these are all existing ones
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Options/LocalUserProfileStorageLocation.cs,4a1c4141d885f30f,references

no reason to not follow established pattern we have.

Reply:
> no reason to not follow established pattern we have.

It hinders our ability to remove it in the future and loses the tight locality the current implementation has.

Reply:
regkey was never a public promise we make. put let customer tweak it as they want if they don't like what we decided. fine we remove it in next release.

Reply:
I don't get that you always block someones PR saying it must follow pattern and now you are not. if you believe it won't get changed by users, then why worry about we might not be able to remove it in the future?

Reply:
The value impacts the code generated for Windows Forms. If it's not consistent, users with different values could see enormous code churn for small designer changes depending on who opened the file last.